controllers/domachine_controller.go

```diff
@@ -194,6 +195,30 @@ func (r *DOMachineReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reterr
 	return r.reconcile(ctx, machineScope, clusterScope)
 }
 
+func (r *DOMachineReconciler) reconcileVolumes(ctx context.Context, mscope *scope.MachineScope, cscope *scope.ClusterScope) (reconcile.Result, error) {
+	mscope.Info("Reconciling DOMachine Volumes")
+	computesvc := computes.NewService(ctx, cscope)
+	domachine := mscope.DOMachine
+	for _, disk := range domachine.Spec.DataDisks {
+		volName := infrav1.DataDiskName(domachine, disk.NameSuffix)
+		vol, err := computesvc.GetVolumeByName(volName)
+		if err != nil {
+			if errors.As(err, &computes.TemporaryError{}) {
```

Comment:
why we need custom error type here? isn't returning the usual error enough
`controller-runtime` will keep requeue if there is an error in return even without specify `reconcile.Result{Requeue: true}` afaik

Reply:
gotcha, thx. Will fix that.